flang/lib/Lower/OpenMP.cpp

```diff
@@ -195,6 +187,51 @@ genOMP(Fortran::lower::AbstractConverter &converter,
         privateClauseOperands, firstprivateClauseOperands, sharedClauseOperands,
         copyinClauseOperands,
         procBindClauseOperand.dyn_cast_or_null<StringAttr>());
+    // Handle attribute based clauses.
+    for (const auto &clause : parallelOpClauseList.v) {
+      if (const auto &defaultClause =
+              std::get_if<Fortran::parser::OmpDefaultClause>(&clause.u)) {
+        switch (defaultClause->v) {
+        case Fortran::parser::OmpDefaultClause::Type::Private:
+          parallelOp.default_valAttr(firOpBuilder.getStringAttr(
+              omp::stringifyClauseDefault(omp::ClauseDefault::defprivate)));
+          break;
+        case Fortran::parser::OmpDefaultClause::Type::Firstprivate:
+          parallelOp.default_valAttr(
+              firOpBuilder.getStringAttr(omp::stringifyClauseDefault(
+                  omp::ClauseDefault::deffirstprivate)));
+          break;
+        case Fortran::parser::OmpDefaultClause::Type::Shared:
+          parallelOp.default_valAttr(firOpBuilder.getStringAttr(
+              omp::stringifyClauseDefault(omp::ClauseDefault::defshared)));
+          break;
+        case Fortran::parser::OmpDefaultClause::Type::None:
+          parallelOp.default_valAttr(firOpBuilder.getStringAttr(
+              omp::stringifyClauseDefault(omp::ClauseDefault::defnone)));
+          break;
+        }
+      }
+      if (const auto &procBindClause =
+              std::get_if<Fortran::parser::OmpProcBindClause>(&clause.u)) {
+        switch (procBindClause->v) {
```

Comment:
This could have been the following. The EnumToString function currently returns a string with the first letter in Caps. Currently, the proc_bind_value expected is a string with all letters in lowercase. Hence it cannot be used directly.
Due to https://bugs.llvm.org/show_bug.cgi?id=47225, we are planning to move all String attributes to have first letter with Caps. This will also solve the issue for the default clause where currently the "def" prefix is used. But the code is not there yet. So this change is fine for now. Alternatively, you can make the Caps change in MLIR and use the following and a similar one for the default clause.
```
parallelOp.proc_bind_valAttr(firOpBuilder.getStringAttr(Fortran::parser::OmpProcBindClause::EnumToString(procBindClause->v)));
```

Reply:
Thanks for bringing this up. It came to me when I was writing those helper functions. Attributes naming in `OpenMP` Dialect is a bit different then other existing Dialects(i.e They follow `camelCase` consistently(helps in avoiding above problem)).
https://github.com/llvm/llvm-project/blob/master/mlir/include/mlir/Dialect/GPU/GPUOps.td#L632
This should not affect the pretty printing/parsing part. IMHO it's okay/desirable to print attributes in an `OpenMP` fashion.
I'm OOO till Tuesday, plan to work this after that.